closed/make/Main.gmk

```diff
@@ -27,6 +27,11 @@ CLEAN_DIRS += vm
 	test-image-openj9 \
 	#
 
+ifeq (,$(BUILD_JAVA_SPEC_HEADERS)) 
+  $(shell m4 -D JAVA_SPEC_VERSION=$(VERSION_FEATURE) $(OPENJ9_TOPDIR)/runtime/include/jvmti.h.m4 > $(OPENJ9_TOPDIR)/runtime/include/jvmti.h)
+  BUILD_JAVA_SPEC_HEADERS := "true"
+endif
```

Comment:
I believe this belongs in openj9 and should be in a make rule rather than executed as the makefile is read (which will at least trigger rebuilds of everything that depends on jvmti.h).

Reply:
jvmti.h is required for building target java.base.copy which is prior to targets in openj9. I tried to add a new target in openj9 but it seems doesn't work.

Reply:
I even tried to add the dependency at CMAKELIST.txt(before building target sourcetool) in the root directory in openj9, but it failed when built with Cmake. If it build by UMA, adding a target in GNUMakefile in openj9/runtime solves the issue but cmake is still a problem.

Reply:
You'll want to update `closed/make/copy/Copy-java.base.gmk`, adding a rule to run `m4` to generate `jvmti.h` (within `$(OUTPUTDIR)` somewhere) and adjust the rule
```
$(INCLUDE_TARGET_DIR)/jvmti.h : $(TOPDIR)/openj9/runtime/include/jvmti.h
```
so it depends instead on that file generated by `m4`.

Reply:
I added this dependency before, target java.base.copy is only one case. I tried to build it by UMA in my local machine, java.base.copy is built later. j9nonbuilder.h also needs jvmti.h and it is built before java.base.copy. So I add a same target in runtime/GNUmakefile, which means we have two parts for the same purpose. I also tried to build it on Jenkins, it seems it need jvmti.h before target build-openj9 or java.base. I added dependency to these two targets but no one works. I guess jvmti.h is needed just after configuration.

Reply:
That is not only one target needs jvmti.h, so we have to generate it early.

Reply:
OpenJDK generates jvmti.h so we should be able to do likewise.

Reply:
Perhaps what is needed is an adjustment to `closed/make./Main.gmk` (lines 46-48) to this:
```
java.base-libs : java.base-copy

j9vm-build : java.base-copy
```
This will force java.base-copy to build before j9vm-build. If java.base-copy ensures jvmti.h is generated and available, all should be good.

Reply:
I changed the rule to `java.base-libs : j9vm-build`, otherwise it won't compile
If the rule added depends on the file generated by m4, we have to create directory vm/include first.
I think we might have two methods here:
1. generate file in source code openj9/runtime/include, then remove it when clean it.
2. create a new directory, but it means we will create an existing directory later. But this part of code is in extension repo/make, I'm not sure if I have right to modify it.